tools/mo/openvino/tools/mo/back/ConvolutionNormalizer.py

```diff
@@ -29,22 +33,69 @@ def resolve_convolution_with_group(node: Node, group: int, ir_version: str):
         if weights_shape[0] == node.output:
             # weights are already is in [G*O I X Y] format
             return
-        new_shape = shape_array([node.output, -1, *weights_shape[2:]])
+
+        num_spatial_dims = len(weights_shape[2:])
+        # Reshape has special_zero=True, if zeros are set then original shapes are copied
+        zeros_to_copy_spatial_dims = np.zeros(num_spatial_dims)
+        new_shape = shape_array([node.output, -1, *zeros_to_copy_spatial_dims])
+        reshape = create_op_node_with_second_input(node.graph, Reshape, new_shape,
+                                                   {'override_output_shape': True})
     elif ir_version == 'V10':
-        # TODO rewrite this transformation to generate a shape-computing sub-graph. Ticket 62076
         I = input_shape[1]
-        new_shape = shape_array([group, node.output // group, I // group, *weights_shape[2:]])
-        assert is_fully_defined(weights_shape[2:]) and is_fully_defined(I) and \
-               np.prod(weights_shape) == np.prod(new_shape), 'Initial weights shape {}, grouped weights shape {}' \
-                                                             ''.format(weights_shape, new_shape)
+        if is_fully_defined(weights_shape[2:]) and is_fully_defined(I):
```

Comment:
not clear why these changes are needed. Please give description for this transformation and share sub-graphs before and after

Reply:
These changes are needed if GoupedConvolution weights input has dynamic spatial shapes. 
For GroupedConv we need to reshape inputs from `[N, C_IN, Z, Y, X]` into `[N, GROUPS * C_IN, Z, Y, X]` and if `X, Y, Z` values are unknown we need to retrieve them from `ShapeOf`. 

https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/convolution/Convolution_1.md?plain=1#L80-L81

https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/convolution/GroupConvolution_1.md?plain=1#L58-L59

We have convolutions with undefined spatial shapes but we don't have GroupedConvolutions with undefined spatial shapes, but if someone tries to convert such models we should be able to process that correctly. This subgraph is inserted only if  any of `X, Y, Z` is undefined and it prepares shape value `[N, GROUPS * C_IN, Z, Y, X]` with the use of `ShapeOf, StridedSlice, Gather and Concat` ops.

At the moment i was not able to get brief picture of subgraph i have only huge. I will try to simplify picture and share asap

Reply:
**Before**
![image](https://user-images.githubusercontent.com/5703039/150401838-2915c082-35dc-4516-a3e6-60062cfc21e2.png)

**After** Reshape has weights layout `[N, GROUPS * C_IN, Z, Y, X]`
![image](https://user-images.githubusercontent.com/5703039/150401675-33ba70e2-05f0-403f-8708-f535cc11fb26.png)